lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -482,14 +516,146 @@ namespace Intl
         error = UErrorCode::U_ZERO_ERROR;
 
         written = uloc_toLanguageTag(defaultLocaleId, bcp47, ULOC_FULLNAME_CAPACITY, true, &error);
-        ICU_ASSERT_RETURN(error, written > 0 && written < cchLangtag, 0);
+        ICU_ASSERT_RETURN(error, written  > 0 && ((size_t) written) < cchLangtag, 0);
 
         return utf8::DecodeUnitsIntoAndNullTerminateNoAdvance(
             langtag,
             reinterpret_cast<LPCUTF8>(bcp47),
             reinterpret_cast<LPCUTF8>(bcp47 + written)
         );
     }
+
+    bool CollatorGetCollation(_In_z_ const char *langtag, _Out_ char *collation, _In_ size_t cchCollation)
+    {
+        Assert(langtag != nullptr && collation != nullptr && cchCollation >= 0 && cchCollation < ULOC_FULLNAME_CAPACITY);
```

Comment:
And if something is distressingly wrong, what would that mean? Shouldn't we failfast?

---
In reply to: [149827118](https://github.com/Microsoft/ChakraCore/pull/4083#discussion_r149827118) [](ancestors = 149827118)

Reply:
Maybe best not to just consider what "should' happen. We're also looking to code defensively. Some sort of stack corruption could possibly lead us here with bad inputs.

We're asserting here because we really don't want to continue if the inputs were unexpected. Is it possible to continue in a reasonable way if they are wrong? What is the real cost of a fail-fast check?

Yes, a failfast check adds release runtime overhead. But this is not a perf-critical path and we prefer to failfast unless there's a good reason not to.

---
In reply to: [149827237](https://github.com/Microsoft/ChakraCore/pull/4083#discussion_r149827237) [](ancestors = 149827237,149827118)